proto/v1beta1/run_function.proto

```diff
@@ -63,6 +63,19 @@ message RunFunctionRequest {
   // and that context will be passed to subsequent Functions. Crossplane
   // discards all context returned by the last Function in the pipeline.
   optional google.protobuf.Struct context = 5;
+
+  // Optional extra resources that the Function required.
+  // Note that extra resources is a map to Resources, plural.
+  // The map key corresponds to the key in a RunFunctionResponse's
+  // extra_resources field. If a Function requested extra resources that
+  // did not exist, Crossplane sets the map key to an empty Resources message to
+  // indicate that it attempted to satisfy the request.
+  map<string, Resources> extra_resources = 6;
+}
+
+// Resources just exists because you can't have map<string, repeated google.protobuf.Struct>.
```

Comment:
```suggestion
// Resources represents the state of a several Crossplane resources.
```

I know I wrote the original comment here, but I think we should update it to something more meaningful to readers who aren't designing the API.